### PR TITLE
middleware: increase code coverage of jwt error condition

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Cover a previously uncovered error condition.

Signed-off-by: John Mulligan <jmulligan@redhat.com>

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
@centos-ci ok to test

File: middleware/jwt_test.go [hunk 1]
```diff
@@ -526,3 +529,68 @@ func TestJwtInvalidKeys(t *testing.T) {
...
+	tests.Assert(t, err == nil, "http.NewRequest failed:", err)
+
+	// Miss 'bearer' string
+	req.Header.Set("Authorization", "bearer "+tokenString)
+	r, err := http.DefaultClient.Do(req)
+	tests.Assert(t, err == nil, "http.DefaultClient failed:", err)
+	tests.Assert(t, r.StatusCode != 0)
+	s, err := utils.GetStringFromResponse(r)
+	tests.Assert(t, strings.Contains(s, "Unexpected signing method"),
+		`expected s to contain "Unexpected signing method", got:`, s)
```
Comment (raghavendra-talur) [inline middleware/jwt_test.go]: @phlogistonjohn Is the backquote here right? I don't know what it does.

**Comment by phlogistonjohn** [comment]
The backticks create a [raw string literal](https://golang.org/ref/spec#String_literals) which is generally for strings where the language doesn't interpret the contents and is used in this case as a cheap way to avoid escaping the double quotes around the expected string.

Let me know if you or the team has a strong preference against this use because I have a lot of upcoming changes that use it frequently. :-)

File: middleware/jwt_test.go [hunk 2]
```diff
@@ -526,3 +529,68 @@ func TestJwtInvalidKeys(t *testing.T) {
...
+	pk, err := rsa.GenerateKey(rand.Reader, 256*2)
+	tests.Assert(t, err == nil, "rsa.GenerateKey failed:", err)
+	tokenString, err := token.SignedString(pk)
+	tests.Assert(t, err == nil, "token.SignedString failed:", err)
+
+	// Setup header
+	req, err := http.NewRequest("GET", ts.URL, nil)
+	tests.Assert(t, err == nil, "http.NewRequest failed:", err)
+
+	// Miss 'bearer' string
```
Comment (obnoxxx) [inline middleware/jwt_test.go]: I think this is a copy'n'paste of a copy'n'paste error, right? We *do* add the bearer string...

I.e. if my understanding is correct,  this comment should be removed. :-)

**Comment by phlogistonjohn** [comment]
Updated with feedback incorporated.

**Comment by obnoxxx** [comment]
LGTM. the centos-ci failure is unrelated.

**Pull request merged**